pkg/project/sync.go

```diff
@@ -286,43 +301,28 @@ func syncFiles(projectPath string, projectID string, conURL string, synctime int
 	}
 
 	if errText != "" {
-		return fileList, directoryList, modifiedList, uploadedFiles, &ProjectError{errOpSyncRef, errors.New(errText), errText}
+		return &SyncInfo{fileList, directoryList, modifiedList, uploadedFiles}, &ProjectError{errOpSyncRef, errors.New(errText), errText}
 	}
 
-	return fileList, directoryList, modifiedList, uploadedFiles, nil
+	return &SyncInfo{fileList, directoryList, modifiedList, uploadedFiles}, nil
 }
 
-func completeUpload(projectID string, files []string, directories []string, modfiles []string, conID string, currentSyncTime int64) (string, int) {
-
-	conInfo, conInfoErr := connections.GetConnectionByID(conID)
-	if conInfoErr != nil {
-		return conInfoErr.Desc, 1
-	}
-
-	conURL, conErr := config.PFEOriginFromConnection(conInfo)
-	if conErr != nil {
-		return conErr.Desc, 1
-	}
+func completeUpload(client utils.HTTPClient, projectID string, completeRequest CompleteRequest, conInfo *connections.Connection, conURL string) (string, int) {
 
 	uploadEndURL := conURL + "/api/v1/projects/" + projectID + "/upload/end"
-	payload := &CompleteRequest{FileList: files, DirectoryList: directories, ModifiedList: modfiles, TimeStamp: currentSyncTime}
-	jsonPayload, _ := json.Marshal(payload)
+	jsonPayload, _ := json.Marshal(&completeRequest)
 	req, err := http.NewRequest("POST", uploadEndURL, bytes.NewBuffer(jsonPayload))
-	req.Header.Set("Content-Type", "application/json")
 	if err != nil {
 		fmt.Printf("error setting the header  %v\n", err)
```

Comment:
This is an error creating the request, not setting the header. (And it never was, the error message was wrong before you changes.)